invenio/modules/oauthclient/contrib/orcid.py

```diff
@@ -122,12 +122,43 @@ def account_info(remote, resp):
 
 def account_setup(remote, token):
     """Perform additional setup after user have been logged in."""
+    from invenio.ext.cache import cache
+    from invenio.modules.accounts.models import UserEXT
     from invenio.ext.sqlalchemy import db
     from ..handlers import token_session_key
 
+    import json
+    import requests
+
+    orcid = session.get(token_session_key(remote.name) +
+                        "_account_info").get("external_id")
+
     extra_data = {
-        "orcid": session.get(token_session_key(remote.name) +
-                             "_account_info").get("external_id")
+        "orcid": orcid
         }
     token.remote_account.extra_data = extra_data
     db.session.commit()
+
+    # Fill user full name if not already set
+    user = UserEXT.query.filter_by(id=orcid).first().user
+    if user and not any([user.given_names, user.family_name]):
+        # Query ORCID to get the real name
+        request_url = 'http://orcid.org/{}/orcid-bio'.format(orcid)
```

Comment:
@lnielsen 
@jmartinm (BTW, thanks for calling me to this thread!)

I'm working on a wrapper, that's true.
The project mentioned by Lars is not longer maintaned, and it doesn't work with current API.

My project is available here:
https://github.com/MSusik/python-orcid

So far it's under development, some things work, some don't. Well, it was under development. We were talking with the ORCID team and they are planning to release a new API in few months which won't be backward compatible, so there is no point in developing now. I'm planning to cover the whole API, both pulling and pushing data.

If you want contribute, it would be great! I'll call you when the new API is ready and I will be working back on the project.